### PR TITLE
[ip6] set `Mesh Local IID` for Reference Devices

### PR DESCRIPTION
**Pull request by simonlingoogle**

### CODE REVIEW

File: src/cli/cli.hpp [hunk 1]
```diff
@@ -736,6 +739,9 @@ class Interpreter
         {"macfilter", &Interpreter::ProcessMacFilter},
 #endif
         {"masterkey", &Interpreter::ProcessMasterKey},
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+        {"mliid", &Interpreter::ProcessMlIid},
```
Comment (abtink) [inline src/cli/cli.hpp:743]: Add documentation for new command to `cli/README.md`?

File: src/core/thread/mle.hpp [hunk 2]
```diff
@@ -348,6 +348,21 @@ class Mle : public InstanceLocator, private NonCopyable
...
 
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+    /**
+     * This method sets the Mesh Local IID.
+     *
+     * Available only when `OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE` is enabled.
+     *
+     * @param[in] aMlIid  The Mesh Local IID.
+     *
+     * @retval OT_ERROR_NONE            Successfully configured Mesh Local IID.
```
Comment (abtink) [inline src/core/thread/mle.hpp:359]: `OT_ERROR_NOTE` -> `kErrorNone`
Same for other error.

**Review by abtink — APPROVED** [review]
LGTM. 👍 
Two suggestions below.

File: include/openthread/ip6.h [hunk 3]
```diff
@@ -719,6 +719,20 @@ otError otIp6RegisterMulticastListeners(otInstance *
                                         otIp6RegisterMulticastListenersCallback aCallback,
                                         void *                                  aContext);
 
+/**
+ * This function sets the Mesh Local IID (for test purpose).
+ *
+ * Only available when `OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE` is enabled.
+ *
+ * @param[in]   aInstance   A pointer to an OpenThread instance.
+ * @param[in]   aIid        A pointer to the Mesh Local Iid to set.
```
Comment (jwhui) [inline include/openthread/ip6.h:728]:
```suggestion
 * @param[in]   aIid        A pointer to the Mesh Local IID to set.
```

File: tests/scripts/thread-cert/test_set_mliid.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        self.nodes[LEADER].set_mliid('1122334455667788')
+
+        self.nodes[LEADER].start()
+        self.simulator.go(4)
+        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+
+        mleid = self.nodes[LEADER].get_ip6_address(config.ADDRESS_TYPE.ML_EID)
+        self.assertEqual(b'\x11\x22\x33\x44\x55\x66\x77\x88', ipaddress.IPv6Address(mleid).packed[-8:])
+
+        self.assertRaises(Exception, lambda: self.nodes[LEADER].set_mliid('5566778811223344'))
```
Comment (librasungirl) [inline tests/scripts/thread-cert/test_set_mliid.py:58]: How about also verifying the configured mliid persistent after reset (not factory reset)?

**Review by librasungirl — APPROVED** [review]

File: tests/scripts/thread-cert/test_set_mliid.py (hunk 4 above)
Comment (simonlingoogle) [reply tests/scripts/thread-cert/test_set_mliid.py:58]: I think it's not required so. The `migrateNetwork` API will get the current ML-IID first, and then set ML-IID after reset.

Comment (librasungirl) [reply tests/scripts/thread-cert/test_set_mliid.py:58]: The original suggestion has nothing to do with future `migrateNework() THCI`, it is just a functional test for manual mliid - mliid should be stored in non-volatile storage once attached, and should be restored if reboot; this test verifies the manual miiid take effects, maybe good to verify it still takes effort after reboot for completeness.

Comment (simonlingoogle) [reply tests/scripts/thread-cert/test_set_mliid.py:61]: > The original suggestion has nothing to do with future `migrateNework() THCI`, it is just a functional test for manual mliid - mliid should be stored in non-volatile storage once attached, and should be restored if reboot; this test verifies the manual miiid take effects, maybe good to verify it still takes effort after reboot for completeness.

SGTM. Misunderstood your intention.

**Comment by jwhui** [comment]
On the latest push, test_set_mliid.py seems to have failed once:
- [logs_76160.zip](https://github.com/openthread/openthread/files/6462037/logs_76160.zip)
- [packet-verification-1.1-on-1.2-pcaps.zip](https://github.com/openthread/openthread/files/6462038/packet-verification-1.1-on-1.2-pcaps.zip)

Thoughts?

**Comment by simonlingoogle** [comment]
> On the latest push, test_set_mliid.py seems to have failed once:
> 
> * [logs_76160.zip](https://github.com/openthread/openthread/files/6462037/logs_76160.zip)
> * [packet-verification-1.1-on-1.2-pcaps.zip](https://github.com/openthread/openthread/files/6462038/packet-verification-1.1-on-1.2-pcaps.zip)
> 
> Thoughts?

That's pretty strange because I can `PASS 1000 FAIL 0` locally.
According to the log, the Leader failed to become Leader again 10s after reset .

**Review by jwhui — APPROVED** [review]

**Pull request merged**